Added pkg-config stuff to cmake scripts

ik-wg opened the pull request:
Hi!

These are minimal changes required to "make install" for ppconsul

One thing i didnt like is making pkg-config to install json11.a into users system.

oliora commented on ext/json11/CMakeLists.txt:
```diff
@@ -4,6 +4,38 @@
+endif(NOT DEFINED CMAKE_INSTALL_LIBDIR)
+set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
+
+set_target_properties(json11 PROPERTIES DEBUG_POSTFIX -d FOLDER json11)
```
Why do you need to set `DEBUG_POSTFIX`?

ik-wg replied:
Also not really needed.

oliora commented on ext/json11/CMakeLists.txt:
```diff
@@ -4,6 +4,38 @@
+endif(NOT DEFINED CMAKE_INSTALL_LIBDIR)
+set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
+
+set_target_properties(json11 PROPERTIES DEBUG_POSTFIX -d FOLDER json11)
```
Here and below in this file use `${PROJECT_NAME}` instead of harcoded `json11`.

ik-wg replied:
OK.

ik-wg commented on CMakeLists.txt:
```diff
@@ -81,5 +86,12 @@ include_directories(include)
 
 add_subdirectory(ext/json11)
-add_subdirectory(src)
+add_subdirectory(ppconsul)
```
Not really needed.

ik-wg commented on ext/json11/CMakeLists.txt:
```diff
@@ -4,6 +4,38 @@
+endif()
+
+install(
+    TARGETS ${install_libs}
```
OK.

ik-wg commented on ppconsul/ppconsul.pc.in:
```diff
@@ -0,0 +1,6 @@
+# ppconsul pkg-config source file
+
+Name: ppconsul
+Description: ppconsul is a consul interface library inplemented in c++
```
OK.

ik-wg commented:
I have addressed most of the feedback in https://github.com/oliora/ppconsul/pull/14/commits/14c2a1d7d47518808c75d7b463916f5d32dcbf3f
please check it out.

re: json11 

For now it only builds static libs. I`d leave it as is for now and make an issue in the tracker to update build scripts to support shared libs config, should be fixed together with it.

oliora commented:
Merged
